Library/Homebrew/test/Gemfile

```diff
@@ -2,6 +2,7 @@ source "https://rubygems.org"
 
 require_relative "../constants"
 
+gem "aws-sdk-s3"
```

Comment:
Can the relevant objects be stubbed/mocked rather than needing to add the gem at test time?

Reply:
Yes, instead of turning on stubs for the S3 library I could instead mock the S3 library interface itself, which would no longer exercise any of the S3 library itself during testing.

Reply:
Yeh, that sounds like a good call for this case, thanks!